spec/models/series_spec.rb

```diff
@@ -26,63 +26,63 @@
-        Psicologia tradicional e a Quântica. Ele sua fala ele mostra que o
-        modelo comportamental',
-      category: category
-    )
-  end
-
-  let!(:video) do
-    Video.create!(
-      name: 'foo bar',
-      description: 'Amig Goswami é uma das maiores autoridades mundiais em
-        Psicologia Quântica. Nesta palestra, ele faz um contraponto entre a
-        Psicologia tradicional e a Quântica. Ele sua fala ele mostra que o
-        modelo comportamental',
-      vimeo_id: '163721649',
-      series: series2,
-      image_url: 'http://foobar.png'
-    )
-  end
-
   describe 'associations' do
-    it { expect(subject).to belong_to(:category) }
-    it { expect(subject).to have_many(:videos) }
+    it { is_expected.to belong_to(:category) }
+    it { is_expected.to have_many(:videos) }
   end
 
   describe 'validations' do
-    it { expect(subject).to validate_presence_of(:category) }
-    it { expect(subject).to validate_presence_of(:name) }
-    it { expect(subject).to validate_presence_of(:description) }
+    it { is_expected.to validate_presence_of(:category) }
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:description) }
   end
 
-  describe 'get image url' do
+  describe '.image' do
     context 'with series image_url set' do
+      let!(:series) do
+        Series.create!(
+          name: 'foo bar',
+          description: 'Amig Goswami é uma das maiores autoridades mundiais em
+            Psicologia Quântica. Nesta palestra, ele faz um contraponto entre a
+            Psicologia tradicional e a Quântica. Ele sua fala ele mostra que o
+            modelo comportamental',
+          category: category,
+          image_url: 'http://foo'
+        )
+      end
+
       it 'returns the series image url' do
         expect(series.image).to eq('http://foo')
       end
     end
 
     context 'without series image_url set' do
+      let!(:series) do
+        Series.create!(
+          name: 'foo bar',
+          description: 'Amig Goswami é uma das maiores autoridades mundiais em
+            Psicologia Quântica. Nesta palestra, ele faz um contraponto entre a
+            Psicologia tradicional e a Quântica. Ele sua fala ele mostra que o
+            modelo comportamental',
+          category: category
+        )
+      end
+
+      let!(:video) do
+        Video.create!(
+          name: 'foo bar',
+          description: 'Amig Goswami é uma das maiores autoridades mundiais em
+            Psicologia Quântica. Nesta palestra, ele faz um contraponto entre a
+            Psicologia tradicional e a Quântica. Ele sua fala ele mostra que o
+            modelo comportamental',
+          vimeo_id: '163721649',
+          series: series,
+          image_url: 'http://foobar.png'
+        )
+      end
+
       it 'returns the first video image url' do
-        expect(series2.image).to eq(video.image_url)
+        expect(series.image).to eq(video.image_url)
```

Comment:
Isn't it better to set the subject to series.image and expect on the subject? What does rubocop-rspec has to say about that?

Reply:
I got the idea. In this specific case, 'series' is a different object in each context. How to have a single subject in this case?

Reply:
I made some changes in that direction.